recipes/mingw-w64/all/conanfile.py

```diff
@@ -13,29 +13,193 @@ class MingwConan(ConanFile):
     settings = "os", "arch", "compiler"
     options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
     default_options = {"threads": "posix", "exception": "seh"}
-    build_requires = "7zip/19.00"
     no_copy_source = True
 
     def validate(self):
-        if self.settings.os != "Windows":
-            raise ConanInvalidConfiguration("MinGW is only supported by Windows.")
+        valid_os = self.conan_data["sources"][self.version]["url"].keys()
+        if str(self.settings.os) not in valid_os:
+            raise ConanInvalidConfiguration(f"MinGW {self.version} is only supported for the following operating systems: {valid_os}")
+        valid_arch = self.conan_data["sources"][self.version]["url"][str(self.settings.os)].keys()
+        if str(self.settings.arch) not in valid_arch:
+            raise ConanInvalidConfiguration(f"MinGW {self.version} is only supported for "
+                                            f"the following architectures on {str(self.settings.os)}: {valid_arch}")
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            self.build_requires("7zip/19.00")
```

Comment:
This belongs in the method `build_requirements`.
Also, `7zip` is only available on a Windows **build system**.
So you need to do:
```python
@property
def _settings_build(self):
    return getattr(self, "settings_build", self.settings)
def build_requiremenents(self):
    if self._settings_build.os == "Windows":
        self.build_requires("7zip/19.00")
```

Reply:
Will be fixed in upcoming commit.